documentation/design-docs/dotnet-tools.md

```diff
@@ -1,124 +1,797 @@
+
+3. Do we have a stripped down stand-alone collector?
+
+    Not at this time, though I still advocate for good layering. There are at least two useful scenarios for this I could see, one is a persistent monitoring agent and the other is on-demand install by support engineers in response to a production problem. In the latter case size on disk is probably only a small concern relative to being able to isolate and customer confidence in the tools. If we could lose the SDK dependency we'd probably be in decent shape there. In the former case size does matter more, but we again be better optimizing for no SDK dependency and FDD deployment before eliminating the analysis components becomes the top size issue. There are also other non-size concerns if we ultimately go this path such as daemonizing the tool, log cycling, auto-triggered collections and remote administration.
+
+4. Do we support command line response files?
+
+    Not at this time. I didn't see any documentation suggesting that dotnet supports the response file generally and it would be nice to follow suit rather than rolling our own. We can revisit this based on customer feedback.
+
+5. Do we support '/' style args that are more common on windows or only '--' style args?
+
+    Not at this time (feel like a broken record yet?). We should document and parse all arguments accepting only the single dash or double dash form. For example -h or --help are recognized for help, but /help is not. This keeps us identical to the behavior of other dotnet tools.
+    FWIW there might be value in accepting the /arg form of arguments but I'd rather it gets taken up across all dotnet tools or as a generic feature of the command-line parsing library so that we have some degree of standardization. Even if we did start recognizing the /arg form, I still suggest only printing the -/-- forms in the help to prevent clutter.
+
+6. Do we want the tool be 'dotnet' prefixed or use a separate tool name?
+
+    Dotnet prefixed. Our tools have similarities in function and naming to other non-dotnet tools, so including something that evokes 'dotnet' is important to distinguish them. Java established a precendent of putting a 'j' in front of their tool names but if we followed suit ('d' or 'dn' perhaps) I think it would only create confusion with dotnet's convention of using 'dotnet'. We could also try to name the tools something complete different (SOS is one example) and get users to still associate the name with dotnet, but it feels harder and less effective than simply naming 'dotnet' explicitly. Other names appear to be less discoverable, less easily predicted, and if we don't pick catchy names, probably less easily remembered. 
+    
+## Area specific Questions
+
+### dotnet-counters
+
+- Do we need pre-defined counter sets to make some typical cases easier?
```

Comment:
Not sure I agree with this right here. Just like we have pre-defined *profiles* for dotnet trace, I think we'll need this for dotnet counters as well

Reply:
Are there some canonical customers, or canonical customer feedback we could look at to help inform our choice? I didn't encounter customers asking for a counter viewing CLI tool myself, so largely the work on this part is 2nd hand based on what I've heard from your team + its a convenient tool to prove that the counter-emission pipeline works.

Reply:
I'd assume you obviously don't want to see counters not relevant to your workload?

For example, if you're monitoring a WPF app, you wouldn't want to see empty ASP.NET Counters

Reply:
But the more I think about this, you don't need profiles. We could detect based on what modules are loaded, so this could just be an implementation detail of the monitor tool

Reply:
The counter tool takes a list of providers. If the customer was monitoring a WPF app presumably they would not specify an ASP.NET provider. Where I expect profiles to come in is for making more precise subsets of counters in a provider, for example you only want to see the memory related counters in the runtime but not the threading counters. You could do that via explicit enumeration of the counters on the command line, but it would be verbose.
 Given infinite dev time I have no objection to adding the feature, I'm just trying to figure out what the right the amount of effort to invest is given the anticipated customer demand. If I knew what you were looking at in terms of customer feedback and demand it would help me be on the same page.

Reply:
There was a canonical set of perf counters customers used when diagnosing System.Web applications. Just a cursory search yields lists of relevant counters by workload published by almost every APM vendor and that matches up nicely with what we've heard anecdotally.

Most relevant web counters were related IIS/Http.Sys/WebEngine queue aren't applicable any more and add to the that we don't emit counters yet, there isn't real customer data I have for .NET Core. I'm anticipating demand based on prior experiences with System.Web but I have nothing more rigorous to prove it.

Reply:
I'm mostly curious about customer demand to view the data in a simple local client app (as opposed to an APM tool or web-portal). For example do we have groups of customers commenting that they used to use PerfMon for task X and now they can't?

Reply:
I updated the spec to say there was some contention about this. I'm still open to discuss it and I think its a good feature, I just haven't seen evidence that persuaded me this feature must be present to have an MVP. I'm going to try getting some additional customer eyes on this, and in the near future we'll have a Preview. If this is a pain point I expect we'll hear about it.